src/pybind/mgr/dashboard/controllers/auth.py

```diff
@@ -28,33 +29,48 @@ class Auth(RESTController):
         user_data = AuthManager.authenticate(username, password)
         user_perms, pwd_expiration_date, pwd_update_required = None, None, None
-        if user_data:
-            user_perms = user_data.get('permissions')
-            pwd_expiration_date = user_data.get('pwdExpirationDate', None)
-            pwd_update_required = user_data.get('pwdUpdateRequired', False)
-
-        if user_perms is not None:
-            logger.debug('Login successful')
-            token = JwtManager.gen_token(username)
-            token = token.decode('utf-8')
-            cherrypy.response.headers['Authorization'] = "Bearer: {}".format(token)
-            return {
-                'token': token,
-                'username': username,
-                'permissions': user_perms,
-                'pwdExpirationDate': pwd_expiration_date,
-                'sso': mgr.SSO_DB.protocol == 'saml2',
-                'pwdUpdateRequired': pwd_update_required
-            }
+        max_attempt = Settings.ACCOUNT_LOCKOUT_ATTEMPTS
+        if max_attempt == 0 or mgr.ACCESS_CTRL_DB.get_attempt(username) < max_attempt:
+            if user_data:
+                user_perms = user_data.get('permissions')
+                pwd_expiration_date = user_data.get('pwdExpirationDate', None)
+                pwd_update_required = user_data.get('pwdUpdateRequired', False)
 
-        logger.debug('Login failed')
-        raise DashboardException(msg='Invalid credentials',
-                                 code='invalid_credentials',
-                                 component='auth')
+            if user_perms is not None:
+                logger.info('Login successful: %s', username)
+                mgr.ACCESS_CTRL_DB.increment_attempt(username, True)
+                mgr.ACCESS_CTRL_DB.save()
+                token = JwtManager.gen_token(username)
+                token = token.decode('utf-8')
+                cherrypy.response.headers['Authorization'] = "Bearer: {}".format(token)
+                return {
+                    'token': token,
+                    'username': username,
+                    'permissions': user_perms,
+                    'pwdExpirationDate': pwd_expiration_date,
+                    'sso': mgr.SSO_DB.protocol == 'saml2',
+                    'pwdUpdateRequired': pwd_update_required
+                }
+            mgr.ACCESS_CTRL_DB.increment_attempt(username)
+            mgr.ACCESS_CTRL_DB.save()
+        else:
+            try:
+                user = mgr.ACCESS_CTRL_DB.get_user(username)
+                user.enabled = False
+                mgr.ACCESS_CTRL_DB.save()
+                warning_msg = ('Maximum number of unsuccessful log-in attempts '
+                               '(' + str(max_attempt) + ') reached for '
+                               'username "' + username + '" so the account was blocked. '
+                               'An administrator will need to re-enable the account')
+                logging.warning(warning_msg)
```

Comment:
I'd rather keep everything in the same statement and with string interpolation (`%`):
```suggestion
                logging.warning(
                            'Maximum number of unsuccessful log-in attempts (%d) reached '
                            'for username "%s" so the account was blocked.  An administrator'
                            ' will need to re-enable the account',
                            max_attempt, username)
```

Reply:
I don't know why it didnt worked for me earlier, maybe some typo error but now its working. Thanks